_rules/aria-role-permitted-j7zzqr.md

```diff
@@ -0,0 +1,167 @@
+
+This rules apply to every element, even if they are not [included in the accessibility tree][]. This is because the roles of `none` or `presentation` are only allowed on certain elements. If the rule was only looking at content [included in the accessibility tree][], it wouldn't flag incorrect use of these roles. 
+
+### Related rules
+
+- [Role attribute has valid value](https://www.w3.org/WAI/standards-guidelines/act/rules/674b10/proposed/) checks that the value of the `role` attribute exists in ARIA, while this rule checks that it is allowed on the element using it.
+
+### Bibliography
+
+- [Document conformance requirements for use of ARIA attributes in HTML](https://www.w3.org/TR/html-aria/#docconformance)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `a` element has an [explicit role][] of `button`, which is allowed.
+
+```html
+<a href="https://www.w3.org/WAI/standards-guidelines/act/rules/" role="button">All ACT rules</a>
+```
+
+#### Passed Example 2
+
+This `h1` element has an [explicit role][] of `tab`, which is allowed
+
+```html
+<h1 role="tab">ACT rules</h1>
+```
+
+#### Passed Example 3
+
+These `hr` elements have an [explicit role][] of `presentation`, which is allowed
+
+```html
+Fruits:
+<ul>
+  <li>Apple
+    <hr role="presentation"/>
+  </li>
+  <li>Banana
+    <hr role="presentation"/>
+  </li>
+  <li>Orange</li>
+</ul>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `button` element has an [explicit role][] of `heading`, which is not allowed.
+
+```html
+<button role="heading">ACT rules</button>
+```
+
+#### Failed Example 2
+
+This `aside` element has an [explicit role][] of `navigation`, which is not allowed.
+
+```html
+<aside role="navigation">
+  <a href="https://www.w3.org">W3C</a> <a href="https://www.w3.org/TR/WCAG21/">WCAG 2.1</a> <a href="https://www.w3.org/WAI/standards-guidelines/act/rules/"> ACT rules</a>
+</aside>
+```
+
+#### Failed Example 3 
+
+These `h1` elements have an [explicit role][] of `listitem`, which is not allowed; the `div` element has an [explicit role][] of `list`, which is allowed.
+
+```html
+Fruits:
+<div role="list">
+  <h1 role="listitem">Apple</h1>
+  <h1 role="listitem">Banana</h1>
+  <h1 role="listitem">Orange</h1>
+</div>
+```
+
+#### Failed Example 4
+
+The first `li` element has an [explicit role][] of `presentation`, which is not allowed.
+
+```html
+Fruits:
+<ul>
+  <li role="presentation">Apple</li>
+  <li>Banana</li>
+  <li>Orange</li>
+</ul>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+There is no [HTML element][namespaced element].
+
+```svg
+<svg xmlns="http://www.w3.org/2000/svg">
+  <title>This is an SVG</title>
+</svg>
+```
+
+#### Inapplicable Example 2
+
+This `button` element is not [included in the accessibility tree][].
+
+```html
+<button role="list" style="display:none;">Click me</button>
```

Comment:
From the background:

> This rules apply to every element, even if they are not [included in the accessibility tree][].

Wouldn't that make this a failed example instead of an inapplicable example?

Reply:
I agree with @tbostic32 

The applicability are elements with an explicit role defined which is the case.

Reply:
True, moved it to an Failed case.

I think that this is still failing ARIA in HTML, even if it doesn't cause an accessibility problem in the end, so I'd rather keep it that way. ARIA in HTML doesn't specify that the roles are only allowed on elements that are exposed. Moreover, if we restrict the rule to exposed elements, suddenty Failed Example 4 (`<li role="presentation">`) becomes Inapplicable when it's a clear failure since the only allowed role is `listitem`.

Reply:
In accordance with my comment https://github.com/act-rules/act-rules.github.io/pull/2084#discussion_r1340254070, to me, Failed Example 4 is not failing because of the presence of the "presentation" role in a DOM children. Rather, its failure is attributed to the absence of an accessibility child with the role "listitem" within the DOM child.

Hence, it is not the "role=presentation" itself that impacts the test result; rather, the test fails due to the absence of a child with the "listitem" role.

I agree with Tom that we should consider elements included into the accessibility tree.

Reply:
just going to state one more time that i'd err on the side of flagging this - as otherwise it falls on making a developer/tester have to do more work to find errors, by ensuring they fully comb through a UI and reveal all hidden content.  I do acknoweldge there are probably instances of devs hiding problematic markup, rather than fixing it... but should a dev really be surprised to get an error flagged for something they knew was an error and 'hid' it as their 'fix'?